src/app/components/challenge/challengeoverview/challengeoverview.component.spec.ts

```diff
@@ -23,11 +25,54 @@ describe('ChallengeoverviewComponent', () => {
 
   beforeEach(() => {
     fixture = TestBed.createComponent(ChallengeoverviewComponent);
+    globalService = TestBed.get(GlobalService);
+    apiService = TestBed.get(ApiService);
     component = fixture.componentInstance;
+
+    spyOn(globalService, 'showModal');
+    spyOn(globalService, 'showToast');
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('global variables', () => {
+    expect(component.isChallengeHost).toBeFalsy();
+  });
+
+  it('should show modal and successfully edit the challenge description', () => {
+    const updatedDescription = 'Updated challenge description';
+    const expectedSuccessMsg = 'The description is successfully updated!';
+    spyOn(apiService, 'patchUrl').and.returnValue(new Observable((observer) => {
+      observer.next({results: [{}]});
+      observer.complete();
+      return {unsubscribe() {}};
+    }));
+
+    component.editChallengeOverview();
+    expect(globalService.showModal).toHaveBeenCalled();
+    component.apiCall(updatedDescription);
+    expect(apiService.patchUrl).toHaveBeenCalled();
+    expect(globalService.showToast).toHaveBeenCalledWith('success', expectedSuccessMsg, 5);
+  });
+
+  it('should handle the API error for `editChallengeOverview` method', () => {
+    const updatedDescription = 'Updated challenge description';
+    const expectedErrorMsg = {
+      error: 'Api error'
+    };
+    spyOn(apiService, 'patchUrl').and.returnValue(new Observable((observer) => {
+      observer.error({error: expectedErrorMsg.error});
+      observer.complete();
+      return {unsubscribe() {}};
+    }));
+
+    component.editChallengeOverview();
+    expect(globalService.showModal).toHaveBeenCalled();
+    component.apiCall(updatedDescription);
+    expect(apiService.patchUrl).toHaveBeenCalled();
+    expect(globalService.showToast).toHaveBeenCalledWith('error', expectedErrorMsg);
+  });
```

Comment:
Why is this changed in "modified styles.." PR?

Reply:
will remove that